### PR TITLE
change etcd dns to loadbalancer again

### PR DESCRIPTION
**Pull request by teemow**
* add etcd port to api loadbalancer
* change the etcd dns record to the api loadbalancer

### CODE REVIEW

File: service/controller/v19/templates/cloudformation/guest/record_sets.go [hunk 1]
```diff
@@ -20,12 +20,13 @@ const RecordSets = `{{define "record_sets"}}
...
+      AliasTarget:
+        DNSName: !GetAtt ApiLoadBalancer.DNSName
+        HostedZoneId: !GetAtt ApiLoadBalancer.CanonicalHostedZoneNameID
+        EvaluateTargetHealth: false
       Name: 'etcd.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'
       HostedZoneId: !Ref 'HostedZone'
-      TTL: '60'
-      Type: CNAME
-      ResourceRecords:
-        - !GetAtt {{ $v.MasterInstanceResourceName }}.PrivateDnsName
```
Comment (kopiczko) [inline service/controller/v19/templates/cloudformation/guest/record_sets.go]: I miss how we add the master to the LB backend.

Comment (teemow) [reply service/controller/v19/templates/cloudformation/guest/record_sets.go]: This is the API loadbalancer. So this should already be there.

Comment (teemow) [reply service/controller/v19/templates/cloudformation/guest/record_sets.go]: https://github.com/giantswarm/aws-operator/blob/master/service/controller/v19/templates/cloudformation/guest/load_balancers.go#L19

**Comment by teemow** [comment]
:+1: I also started testing this manually already.

**Comment by teemow** [comment]
Thanks.

**Comment by teemow** [comment]
this is ready again. PTAL. 

the upgrade will only be graceful with the follow-up (https://github.com/giantswarm/aws-operator/pull/1288) but this one here is ready.

**Comment by calvix** [comment]
There is one more thing, before this change we already implemented monitoring of tenant cluster etcd and backing of tenant cluster etcd from the control plane. Please make sure that both features work properly with this change.

File: service/controller/v19/adapter/guest_security_groups.go [hunk 2]
```diff
@@ -61,6 +63,9 @@ func (s *GuestSecurityGroupsAdapter) Adapt(cfg Config) error {
 	s.IngressSecurityGroupName = key.SecurityGroupName(cfg.CustomObject, key.KindIngress)
 	s.IngressSecurityGroupRules = s.getIngressRules(cfg.CustomObject)
 
+	s.EtcdELBSecurityGroupName = key.SecurityGroupName(cfg.CustomObject, key.KindEtcdELB)
```
Comment (teemow) [inline service/controller/v19/adapter/guest_security_groups.go]:
```suggestion
	s.EtcdELBSecurityGroupName = key.SecurityGroupName(cfg.CustomObject, key.KindEtcd)
```

File: service/controller/v18patch1/adapter/guest_security_groups.go [hunk 3]
```diff
@@ -184,6 +189,17 @@ func (s *GuestSecurityGroupsAdapter) getIngressRules(customObject v1alpha1.AWSCo
 	}
 }
 
+func (s *GuestSecurityGroupsAdapter) getEtcdELBRules(customObject v1alpha1.AWSConfig) []securityGroupRule {
```
Comment (teemow) [inline service/controller/v18patch1/adapter/guest_security_groups.go]:
```suggestion
func (s *GuestSecurityGroupsAdapter) getEtcdRules(customObject v1alpha1.AWSConfig) []securityGroupRule {
```

File: service/controller/v18patch1/adapter/guest_security_groups.go [hunk 4]
```diff
@@ -61,6 +63,9 @@ func (s *GuestSecurityGroupsAdapter) Adapt(cfg Config) error {
 	s.IngressSecurityGroupName = key.SecurityGroupName(cfg.CustomObject, prefixIngress)
 	s.IngressSecurityGroupRules = s.getIngressRules(cfg.CustomObject)
 
+	s.EtcdELBSecurityGroupName = key.SecurityGroupName(cfg.CustomObject, prefixEtcdELB)
+	s.EtcdELBSecurityGroupRules = s.getEtcdELBRules(cfg.CustomObject)
```
Comment (teemow) [inline service/controller/v18patch1/adapter/guest_security_groups.go]:
```suggestion
	s.EtcdELBSecurityGroupRules = s.getEtcdRules(cfg.CustomObject)
```

File: service/controller/v19/adapter/guest_security_groups.go [hunk 5]
```diff
@@ -61,6 +63,9 @@ func (s *GuestSecurityGroupsAdapter) Adapt(cfg Config) error {
...
+	s.EtcdELBSecurityGroupRules = s.getEtcdELBRules(cfg.CustomObject)
```
Comment (teemow) [inline service/controller/v19/adapter/guest_security_groups.go]:
```suggestion
	s.EtcdELBSecurityGroupRules = s.getEtcdRules(cfg.CustomObject)
```

File: service/controller/v19/adapter/guest_security_groups.go [hunk 6]
```diff
@@ -185,6 +190,17 @@ func (s *GuestSecurityGroupsAdapter) getIngressRules(customObject v1alpha1.AWSCo
 	}
 }
 
+func (s *GuestSecurityGroupsAdapter) getEtcdELBRules(customObject v1alpha1.AWSConfig) []securityGroupRule {
```
Comment (teemow) [inline service/controller/v19/adapter/guest_security_groups.go]:
```suggestion
func (s *GuestSecurityGroupsAdapter) getEtcdRules(customObject v1alpha1.AWSConfig) []securityGroupRule {
```

File: service/controller/v18patch1/adapter/guest_load_balancers.go [hunk 7]
```diff
@@ -22,6 +22,11 @@ type GuestLoadBalancersAdapter struct {
 	APIElbPortsToOpen                []GuestLoadBalancersAdapterPortPair
 	APIElbScheme                     string
 	APIElbSecurityGroupID            string
+	EtcdElbHealthCheckTarget         string
```
Comment (teemow) [inline service/controller/v18patch1/adapter/guest_load_balancers.go]:
```suggestion
	EtcdElbDomain                    string
	EtcdElbHealthCheckTarget         string
```

File: service/controller/v18patch1/adapter/guest_load_balancers.go [hunk 8]
```diff
@@ -50,6 +55,22 @@ func (a *GuestLoadBalancersAdapter) Adapt(cfg Config) error {
...
 	a.APIElbScheme = externalELBScheme
 
+	// etcd load balancer settings.
+	etcdElbName, err := key.LoadBalancerName(key.EtcdDomain(cfg.CustomObject), cfg.CustomObject)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	a.EtcdElbHealthCheckTarget = heathCheckTarget(key.EtcdPort(cfg.CustomObject))
+	a.EtcdElbName = etcdElbName
```
Comment (teemow) [inline service/controller/v18patch1/adapter/guest_load_balancers.go]:
```suggestion
	a.EtcdElbName = etcdElbName
	a.EtcdElbDomain = key.EtcdDomain(cfg.CustomObject)
```

File: service/controller/v19/adapter/guest_load_balancers.go [hunk 9]
```diff
@@ -22,6 +22,11 @@ type GuestLoadBalancersAdapter struct {
 	APIElbPortsToOpen                []GuestLoadBalancersAdapterPortPair
 	APIElbScheme                     string
 	APIElbSecurityGroupID            string
+	EtcdElbHealthCheckTarget         string
```
Comment (teemow) [inline service/controller/v19/adapter/guest_load_balancers.go]:
```suggestion
	EtcdElbDomain                    string
	EtcdElbHealthCheckTarget         string
```

File: service/controller/v19/adapter/guest_load_balancers.go [hunk 10]
```diff
@@ -50,6 +55,22 @@ func (a *GuestLoadBalancersAdapter) Adapt(cfg Config) error {
...
 	a.APIElbScheme = externalELBScheme
 
+	// etcd load balancer settings.
+	etcdElbName, err := key.LoadBalancerName(key.EtcdDomain(cfg.CustomObject), cfg.CustomObject)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	a.EtcdElbHealthCheckTarget = heathCheckTarget(key.EtcdPort(cfg.CustomObject))
+	a.EtcdElbName = etcdElbName
```
Comment (teemow) [inline service/controller/v19/adapter/guest_load_balancers.go]:
```suggestion
	a.EtcdElbName = etcdElbName
	a.EtcdElbDomain = key.EtcdDomain(cfg.CustomObject)
```

**Comment by teemow** [comment]
Tests are fine. etcd elb security group is adapted for the host cluster. Fishing for final LGTM

**Pull request merged**